pandas/core/indexes/multi.py

```diff
@@ -1551,6 +1551,10 @@ def __setstate__(self, state):
 
     def __getitem__(self, key):
         if is_scalar(key):
+            if lib.is_float(key) and key == int(key):
+                # Avoid DeprecationWarning for non-integer number indexer
```

Comment:
we do this a number of times in the indexing code

i think we have a function

maybe_convert_to_int_indexer iirc

if not can u make one

Reply:
Index._convert_scalar_indexer looks like a candidate; need to check where it is called.

Reply:
I take it back.  core.common is right next to `is_bool_indexer` is the place for this.